### PR TITLE
OTA-656: Add CI jobs for heterogeneous cluster upgrades

### PR DESCRIPTION
**Pull request by petr-muller**
- multiarch: add scaffolding for hetero->hetero upgrade jobs
- `make update`

### CODE REVIEW

**Comment by petr-muller** [comment]
It seems we do not have 4.13 heterogeneous releses yet...

**Comment by petr-muller** [comment]
Seeing some noise in the https://steps.ci.openshift.org/reference/ipi-install-heterogeneous step:

```
Creating aarch64 worker MachineSet
Warning: incorrect GroupVersionKind for AWSMachineProviderConfig object: machine.openshift.io/v1beta1, Kind=AWSMachineProviderConfig
machineset.machine.openshift.io/ci-op-8lix2l21-008f2-qtw4c-worker-us-west-2a-additional created
W1104 15:12:13.138293      96 loader.go:223] Config not found: /tmp/installer/auth/kubeconfig
Error from server (Forbidden): routes.route.openshift.io "console" is forbidden: User "system:serviceaccount:ci-op-8lix2l21:ocp-e2e-aws-ovn-heterogeneous-upgrade" cannot get resource "routes" in API group "route.openshift.io" in the namespace "openshift-console"
Exiting with ret=0
```

**Comment by petr-muller** [comment]
1. `nightly-4.11-ocp-e2e-aws-heterogeneous`
  a. `templateinstance readiness test should report ready soon after all annotated objects are ready` does not seem to be related but I want more runs.
  b. Heterogeneous setup looks good
2. `nightly-4.12-upgrade-from-nightly-4.11` looks good
3. `nightly-4.12-upgrade-from-stable-4.11` went well but failed on CI infra flake in post phase
4. `nightly-4.12-ocp-e2e-aws-ovn-heterogeneous`
  a. `alert TelemeterClientFailures fired for 144 seconds with labels: {namespace="openshift-monitoring", severity="warning"}` does not seem to be related but I want more runs
  b. `Warning: incorrect GroupVersionKind for AWSMachineProviderConfig object: machine.openshift.io/v1beta1, Kind=AWSMachineProviderConfig` in a heterogeneous setup step seems to be in all initial 4.12 jobs, and does not seem to be caused by the step itself - this content is pulled from the cluster, modified and re-applied
5. `nightly-4.13-*` jobs all fail with `failed to resolve release arm64-latest: failed to request latest release: server responded with 404: no release configuration exists with the requested name `: there are no 4.13 releases on https://multi.ocp.releases.ci.openshift.org/ yet

File: ci-operator/step-registry/openshift/upgrade/aws/heterogeneous/OWNERS
Comment (petr-muller) [inline ci-operator/step-registry/openshift/upgrade/aws/heterogeneous/OWNERS:4]: I am not sure about a good set of reviewers here, this is a copy paste of others' upgrade workflow OWNERS file.

**Comment by petr-muller** [comment]
/uncc @fabianofranz @jaypoulz 
/cc @deepsm007 @aleskandro @jottofar @wking 

I think is is now reviewable. Several questions:
1. I selected random hours of the day for the cron patterns and set the day to Monday to avoid collisions with https://github.com/openshift/release/pull/31965/, but I do not have the full context about why the crons are actually needed, so I'd welcome insights about whether these are set appropriately
2. Is the scope correct? Do we need these jobs? Fewer of them? Are some missing?
3. Who are good owners for the workflow? OTA? Multiarch folks?

**Comment by deepsm007** [comment]
>     1. I selected random hours of the day for the cron patterns and set the day to Monday to avoid collisions with [MIXEDARCH-37: Arm worker nodes for heterogeneous workflow #31965](https://github.com/openshift/release/pull/31965), but I do not have the full context about why the crons are actually needed, so I'd welcome insights about whether these are set appropriately
--> We can set it to intervals, ultimately it needs to be triggered from release-controller for every nightly

**Comment by petr-muller** [comment]
> --> We can set it to intervals, ultimately it needs to be triggered from release-controller for every nightly

ok, I've changed the jobs to have `interval: 72h` like most other jobs

**Review by deepsm007 — APPROVED** [review]

**Review by aleskandro — APPROVED** [review]

File: ci-operator/step-registry/openshift/upgrade/aws/heterogeneous/OWNERS
Comment (wking) [reply ci-operator/step-registry/openshift/upgrade/aws/heterogeneous/OWNERS:4]: It is probably worth including someone from multi-arch in here, so they can approve new `upgrade-aws-heterogeneous-*` steps/chains/workflows if we need more in the future.  @deepsm007 , would that be you?  Or can you recommend someone else?

I don't think we need to block on having someone from multi-arch though.  If we stick with the parent-dir approver set, `hack/dedup-owners.sh` makes it easy to set up a DRYer symlink.

Comment (deepsm007) [reply ci-operator/step-registry/openshift/upgrade/aws/heterogeneous/OWNERS:4]: @deepsm007 and @Prashanth684 to own heterogeneous

**Review by wking — APPROVED** [review]
We can refresh the LGTM after [the OWNERS tweak][1].

/approve
/hold for OWNERS tweak

[1]: https://github.com/openshift/release/pull/33761#discussion_r1018313734

**Comment by petr-muller** [comment]
/hold cancel
/pj-rehearse

Updated OWNERS, also submitted https://github.com/openshift/release/pull/33932 separately so that I can help with maintaining all of these.

Can I have a re-/lgtm?

**Pull request merged**
